src/main/java/com/google/sps/data/RestrictTimesCriteria.java

```diff
@@ -0,0 +1,53 @@
+package com.google.sps.data;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+public class RestrictTimesCriteria implements Preference {
+  public List<TimeRange> userNoClassTimes;
+  
+  public RestrictTimesCriteria(List<TimeRange> userNoClassTimes) {
+    this.userNoClassTimes = userNoClassTimes;
+  }
+
+  /**
+   * Implements the preference score of the schedule, based on the criteria
+   * that prioritizes the TimeRanges that a user does not want class during
+   */
+  public float preferenceScore(Schedule schedule) {
+    // count the number of minutes that overlap
+    float minutesPenalty = 0;
+    
+    List<Course> courses = (List) schedule.getCourses();
+    List<TimeRange> timeOrderedCourses = getCourseTimes(courses);
```

Comment:
`timeOrderedCourses` -> `courseTimes`?

There's nothing here to indicate that `getCourseTimes()` returns a sorted list, nor that you need it sorted in this code.

Reply:
Changed to `courseTimes`. Thanks for the catch, meant to change this.